### ISSUE TITLE
Can't return null from actions

### ISSUE THREAD

**Issue by molobala** [berkesa/moleculer-java#11]
Impossible to return `null` from actions  
It gives `NullPointerException`  

```
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326) [na:1.8.0_191]
	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338) [na:1.8.0_191]
	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:911) [na:1.8.0_191]
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:953) [na:1.8.0_191]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:926) ~[na:1.8.0_191]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) [na:1.8.0_191]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962) [na:1.8.0_191]
	at io.datatree.Promise.complete(Promise.java:682) ~[datatree-promise-1.0.3.jar:na]
```

**Comment by berkesa** [comment]
Thanks for the comment, I'll see - as soon as possible - what's the problem in this part.

**Comment by molobala** [comment]
i got the problem.  
It was the fact I call `asObject` after `waitFor`being called on the action call like this 

```java
this.broker.call("user.getOne", param).waitFor().asObject()
```
In this case if the action returns null `asObject` fails
So I did like this, adding extra check
```java
Tree ob = this.broker.call("user.getOne", param).waitFor();
User user = ob==null ? null : (User) ob.asObject();

```

**Comment by berkesa** [comment]
Ah, ok, thanks for the info.
Since some other things have changed, I'm about to release a new version soon.
Just a short note: the "waitFor" method is good for testing services. When writing "real" services rather use the non-blockinbg promise syntax, to get the response of a remote service. eg:

```java
broker.call("user.getOne", param).then(rsp -> {
  // rsp is a Tree/JSON
}).catchError(err -> {
  // err is a Throwable
});
```

**Comment by berkesa** [comment]
When we designed the java-based Moleculer, we deliberately avoid using Reflection API (or at least use as little as possible). This is because functions without reflection are much faster. Automatic object mappers (eg. POJO to JSON, or JSON to POJO) can significantly reduce performance. This is the first reason why I recommend using the "Tree" object instead of a return value, instead of a Java Bean (eg. the "User" object).

The other important reason is that "Tree" (~=a JSON structure) will always be "Tree", no matter whether we are using a local or remote function call.

If the "User" type is the return value, the type of the response Class varies depending on where you call the function. If you locally call your action, you will get "User" object, if you call it on a remote node you will receive a "Tree" object. This is because the program serializes the returned value when transmitting it on a network.

The serialized "Tree" will be a similar "Tree" under the deserialization, no matter where we call the method.
It's a little bit longer, but will be faster, and "safer" (returns the same value on a local/remote node):

```java
broker.createService(new Service("user") {

  public Action getOne = ctx -> {

    // Get POJO
    User userPOJO = ...load from DB...

    // Create JSON "directly", without reflection
    Tree userJSON = new Tree();
    userJSON.put("name",    userPOJO.getFullName());
    userJSON.put("address", userPOJO.getMainAddress());
    userJSON.put("phone",   userPOJO.getMobile());
    return userJSON;
  };

});
```

**Comment by molobala** [comment]
I see.
So what if `getOne` action should return a null, is there a method on Tree to encaspulate `null` ?

**Comment by berkesa** [comment]
Yes, in principle this should work. 
In the tests, the "null" value went through between the nodes:

https://github.com/moleculer-java/moleculer-java/blob/master/src/test/java/services/moleculer/service/ServiceTest.java#L92

**Comment by berkesa** [comment]
Fixed in 1.0.7
